corehq/apps/commtrack/models.py

```diff
@@ -410,13 +410,14 @@ def get_display_config(cls):
 
 class ActiveManager(models.Manager):
     """
-    Filter any object that is associated to an archived product.
+    Filter any object that is associated to an archived product, or deleted location.
     """
 
     def get_queryset(self):
         return super(ActiveManager, self).get_query_set() \
             .exclude(sql_product__is_archived=True) \
-            .exclude(sql_location__is_archived=True)
+            .exclude(sql_location__is_archived=True) \
+            .exclude(sql_location=None)
```

Comment:
I talked to Mike O about this and no one has been able to think of a situation where you can have a stock state without a location, so there should not be any harm in this. Since those states are primarily tied to supply point cases, and we keep the cases around this seemed the better option, but I would be happy to delete those too, if that is what people think is better.